src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt

```diff
@@ -901,6 +904,80 @@ class AccountControllerTest @Autowired constructor(
         }
     }
 
+    @NestedTest
+    @DisplayName("PUT /recoverPassword/{recoveryToken}")
+    inner class RecoverPassword {
+        private val newPassword = "new-password"
+
+        private val parameters = listOf(
+            parameterWithName("recoveryToken").description("The recovery token sent to the user's email.")
+        )
+
+        private val passwordRecoveryPayload = PayloadSchema(
+            "password-recover",
+            mutableListOf(
+                DocumentedJSONField("password", "The new password.", JsonFieldType.STRING)
+            )
+        )
+
+        @BeforeEach
+        fun setup() {
+            repository.save(testAccount)
+        }
+
+        @Test
+        fun `should update the password`() {
+            mockMvc.perform(post("/auth/recoverPassword/${testAccount.id}"))
+                .andReturn().response.let { authResponse ->
+                    val recoveryToken = objectMapper.readTree(authResponse.contentAsString)["recovery_url"].asText()
+                        .removePrefix("localhost:8080/accounts/recoverPassword/")
+                    mockMvc.perform(
+                        put("/accounts/recoverPassword/$recoveryToken")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(
+                                objectMapper.writeValueAsString(
+                                    mapOf(
+                                        "password" to newPassword
+                                    )
+                                )
+                            )
+                    ).andExpectAll(
+                        status().isOk()
+                    ).andDocumentCustomRequestSchema(
+                        documentation,
+                        passwordRecoveryPayload,
+                        "Recover password",
+                        "Update the password of an account using a recovery token.",
+                        // urlParameters = parameters,
```

Comment:
It is pretty much ready for review. There is a doubt here because it errors when I document the URL parameters.